mobile/Screens/WritingActivities/ProgressiveWriting.jsx

```diff
@@ -1,19 +1,315 @@
+          <Text>Graphic holder</Text>
+        </View>,
+        <View style={[styles.banner, { height: window.height * 0.35 }]}>
+          <Text>
+            {genreFilter[selectedActivity].activity[step]}
+          </Text>
+        </View>,
+        <TouchableOpacity style={styles.buttonBanner} onPress={() => { setStep(step + 1); }}>
+          <Text>
+            Start
+          </Text>
+        </TouchableOpacity>,
+      );
+      return (
+        content.map((elem) => elem)
+      );
+    } if (step === genreFilter[selectedActivity].activity.length) {
+      const content = [];
+      content.push(
+        <View style={[styles.banner, { height: window.height * 0.2, borderRadius: 10 }]}>
+          <Text>Complete!</Text>
+        </View>,
+        <View style={[styles.banner, { height: window.height * 0.3 }]}>
+          <Text>Graphic holder</Text>
+        </View>,
+        <View style={{
+          display: 'flex',
+          flexDirection: 'row',
+          gap: 10,
+          width: 'auto',
+        }}
+        >
+          <TouchableOpacity title="Save" onPress={() => { console.log('save placeholder'); }} style={[styles.buttonBanner, { width: bannerDim * 0.25 }]}>
+            <Text>
+              Save
+            </Text>
+          </TouchableOpacity>
+          <TouchableOpacity
+            onPress={() => { setStep(0); }}
+            style={[styles.buttonBanner, { width: bannerDim * 0.75 }]}
+          >
+            <Text>Back to Activity List</Text>
+          </TouchableOpacity>
+        </View>,
+      );
+      return (
+        content.map((elem) => elem)
+      );
+    }
+
+    const content = [];
+    content.push(
+      <View style={[styles.banner, { height: window.height * 0.2 }]}>
+        <Text>Graphic holder</Text>
+      </View>,
+      <View style={[styles.banner, { height: window.height * 0.3 }]}>
+        <Text>
+          {genreFilter[selectedActivity].activity[step]}
+        </Text>
+      </View>,
+      <TouchableOpacity title="Next" onPress={() => { setStep(step + 1); }} style={styles.buttonBanner}>
+        <Text>
+          Next
+        </Text>
+      </TouchableOpacity>,
+    );
+    return (
+      content.map((elem) => elem)
+    );
+  };
+
   return (
-    <View style={styles.container}>
-      <Text>Progressive Writing Screen</Text>
+    <View>
+      {
+        genreFilter === null
+          ? (
+            <ScrollView contentContainerStyle={styles.container}>
+              {
+              genreLabels.map((label) => (
+                // 10 Doors Screen
+                <TouchableOpacity
+                  style={styles.activity}
+                  onPress={() => { selectActivityGenre(label); }}
+                >
+                  <Text>
+                    {label}
+                  </Text>
+                </TouchableOpacity>
+              ))
+            }
+            </ScrollView>
+          )
+          : (
+            <View style={{
+              backgroundColor: '#fff',
```

Comment:
Try avoiding inline styling when possible! React Native's StyleSheet.create (like your `styles.banner`) caches styles. But if you declare the styles as inline then during every render your styles are sent over the bridge (which can lessen performance)!